ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/parameter/QueryParameterMapper.java

```diff
@@ -59,8 +57,8 @@ public class QueryParameterMapper extends AbstractParameterMapper {
     private SemanticModel semanticModel;
 
     public QueryParameterMapper(ParameterNode parameterNode, Map<String, String> apiDocs,
-                                OperationInventory operationInventory, Components components,
-                                boolean treatNilableAsOptional, AdditionalData additionalData) {
+                                OperationInventory operationInventory, boolean treatNilableAsOptional,
+                                AdditionalData additionalData, TypeMapper typeMapper) {
```

Comment:
Can't we add this  `treatNilableAsOptional` to the `AdditionalData` object?

Reply:
`AdditionalData` is something we shared among all the mappers but `treatNilableAsOptional` is only applicable for query and header mappers